catboost/python-package/ut/medium/test.py

```diff
@@ -2050,6 +2051,52 @@ def test_get_metadata_notrain():
         assert str(i + 1) in model.get_metadata()
 
 
+# tests for single objects
+
+def get_single_object_from_dataset(pool):
+    test_list = open(data_file(pool, 'test_small')).read().split('\n')[:-1]
+    for i in range(len(test_list)):
+        test_list[i] = test_list[i].split('\t')
+        for j in range(len(test_list[i])):
+            try:
+                test_list[i][j] = float(test_list[i][j])
+            except:
+                test_list[i][j] = test_list[i][j]
+    return random.choice(test_list)
```

Comment:
Maybe instead of reading entire dataset and then selecting one out of it is better to first find out number of lines in a dataset, then choose random line out of it and then parse it?

Also, you are using `np.random.choice` which is not deterministic, and this will result in a flaky tests (basically test are going to be useless), so it's better to fix PRNG seed.

so it can be rewritten as:
```suggestion
     def _get_single_object_from_dataset(name, seed=20181215, rng=None):
         if rng is None:
             rng = np.random.RandomState(seed=seed)
             
         with open(data_file(pool, 'test_small') as f:
             lines = f.read().split('\n')
             
         line = rng.choice(lines)
         parsed = []
         for c in lines.split('\t'):
             try:
                 parsed.append(float(c))
             except:
                 parsed.append(c)
                 
         return parsed
```

Reply:
but instead of wring yet another dataset parser you should probably use `catboost.Pool` instead